otherlibs/unix/write_win32.c

```diff
@@ -39,8 +39,11 @@ CAMLprim value caml_unix_write(value fd, value buf, value vofs, value vlen)
       SOCKET s = Socket_val(fd);
       caml_enter_blocking_section();
       ret = send(s, iobuf, numbytes, 0);
-      if (ret == SOCKET_ERROR) err = WSAGetLastError();
       caml_leave_blocking_section();
+      if (ret == SOCKET_ERROR) {
+        err = WSAGetLastError();
+        if (err == WSAEWOULDBLOCK && written > 0) break;
+      }
```

Comment:
There are guards in place to try to be sure that `caml_leave_blocking_section` doesn't clobber any of the error mechanisms (cf. #10220) but I think it's still probably better to call `WSAGetLastError` before re-acquiring the runtime lock. That would just mean leaving L42 intact and having:
```suggestion
      if (ret == SOCKET_ERROR && err == WSAEWOULDBLOCK && written > 0) break;
```

(otherwise this LGTM!)

Reply:
Amended as suggested. Thanks!